src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -157,5 +160,406 @@ ConvSolution InitAnyInvokerFactory(const ProblemDescriptionType& problem,
     return result;
 }
 
+struct TransposeInstance
+{
+    size_t tensor_sz = 0;
+    std::vector<OpKernelArg> kern_args{};
+    int index         = -1;
+    size_t buf_offset = 0;
+    shared<Data_t> buf_handle{};
+
+    template <typename TransSolnType>
+    TransposeInstance(const TransSolnType& trans_sol, int idx, const MultiBufferWorkspaceTraits& wt)
+    {
+        tensor_sz = trans_sol.GetOutputTensorSize();
+        kern_args = trans_sol.GetKernelArg();
+        index     = idx;
+        assert(index >= 0);
+        buf_offset = wt.GetOffset(index);
+    }
+
+    void AllocBuffer(const Handle& handle, Data_t workSpace)
```

Comment:
[R] "Alloc" implies "Free" but we do not have it. Maybe "GetSubBuffer" or just "SubBuffer"?

Reply:
fixed in #2461